src/dbnode/persist/fs/read_decode_stream.go

```diff
@@ -41,7 +41,7 @@ type readerDecoderStream struct {
 	bytesReader      *bytes.Reader
 	readerWithDigest digest.ReaderWithDigest
 	backingBytes     []byte
-	buf              [8]byte
+	buf              [64]byte
```

Comment:
it looks like your result matches expected cache line size, maybe worth setting this to `len(cpu.CacheLinePad)` from x/sys?

Reply:
Thanks for pointing me to this interesting feature, was not aware of it.
However, I see that this value can potentially be 0 (see `const cacheLineSize = 0` in cpu_wasm.go) which would elevate the issue from performance to correctness one. Also, all the practical target architectures (intel + arm64) have it at 64, and I have no way to benchmark the effect on anything else, so I guess I'll just keep it simple and stick to a fixed number.

Reply:
gotcha!